### PR TITLE
add domain checker

### PR DESCRIPTION
**Pull request by marandaneto**
_#skip-changelog_

The goal of this draft PR is to do this: https://github.com/dart-lang/pub/issues/3307#issuecomment-1063855219

### CODE REVIEW

File: check_domain_access.dart [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    print('- iteration #${i + 1}/$iterations...');
+    await _checkUrls(client: client);
+  }
+  client.close();
+
+  print('Done.');
+  exit(0);
+}
+
+final timeLimit = Duration(seconds: 15);
```
Comment (philipphofmann) [inline check_domain_access.dart:29]: `l`: Is this some random number, or did you choose this with intention? If the latter, we could add a comment about the reasoning.

File: check_domain_access.dart [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+final urls = <String>[
+  // package info
+  'https://pub.dev/api/packages/http',
+  'https://pub.dartlang.org/api/packages/http',
+  // package archive
+  'https://pub.dev/packages/http/versions/0.13.1.tar.gz',
```
Comment (philipphofmann) [inline check_domain_access.dart:35]: `m`: Is it fine to pin this to a fixed version?

Comment (philipphofmann) [inline check_domain_access.dart:35]: `m`: Is it fine to pin this to a fixed version?

**Review by philipphofmann — COMMENTED** [review]
Some minor comments.

**Comment by marandaneto** [comment]
@philipphofmann this is not intended to be merged, the class is copied from the Dart SDK.
It's just a repro case for the Dart team, See PR description.
Thanks for the review anyway.

**Pull request closed without merging**